### PR TITLE
1602 by Inlead: Implement ding_item_viewer into DDB CMS.

### PR DESCRIPTION
**Pull request by inlead**
http://platform.dandigbib.org/issues/1602

### CODE REVIEW

**Comment by guddo** [comment]
This PR is made for the 7.x-2.5.1 release.

File: ding2.info [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name = Ding 2
 description = Install a Ding based library site.
+version = 7.x-2.5.1
```
Comment (cableman) [inline ding2.info]: Version should not be part of this PR.

File: modules/ding_item_viewer/ding_item_viewer.info [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name = Ding Item Viewer
+description = Display ting objects in a carousel view.
+core = 7.x
+package = easyDDB
+version = 7.x-2.2.0+easy
```
Comment (cableman) [inline modules/ding_item_viewer/ding_item_viewer.info]: Package should be Ding and version should not be here.

File: modules/ding_item_viewer/ding_item_viewer.install [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+/**
+ * Implements hook_install().
+ */
+function ding_item_viewer_install() {
+  module_enable(array('ding_more_info'));
+
+  // Create field group.
+  $group = (object) array(
+    'identifier' => 'group_data|ting_object|ting_object|ding_item_viewer_big',
```
Comment (cableman) [inline modules/ding_item_viewer/ding_item_viewer.install]: Why are ting_object there twice in the string?

File: modules/ding_item_viewer/ding_item_viewer.module [hunk 4]
```diff
@@ -0,0 +1,746 @@
...
+      'data' => array(
+        drupal_render($form['ting_searches'][$id]),
+        drupal_render($sort),
+      ),
+      'class' => array('draggable'),
+    );
+  }
+
+  $output = theme_table(array(
+                          'header' => $header,
```
Comment (cableman) [inline modules/ding_item_viewer/ding_item_viewer.module]: Indention is wrong here.

File: modules/ding_item_viewer/ding_item_viewer.module [hunk 5]
```diff
@@ -0,0 +1,746 @@
...
+                          'rows' => $rows,
+                          'attributes' => array('id' => 'ding-item-viewer-queries'),
+                        ));
+
+  $output .= drupal_render_children($form);
```
Comment (cableman) [inline modules/ding_item_viewer/ding_item_viewer.module]: Why not return an render array?

**Comment by cableman** [comment]
Needs a README that explains what the module is used for.

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 3b83715ee9545555ca72b558bcd431fb507b7849
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: eaae3b735cc570703c87f1f26cba6da7843a0f65
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: f27da5bd4475d29085537891a757e04d8afbb5c3
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 3b83715ee9545555ca72b558bcd431fb507b7849
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: eaae3b735cc570703c87f1f26cba6da7843a0f65
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: f27da5bd4475d29085537891a757e04d8afbb5c3
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ding_item_viewer/ding_item_viewer.install (hunk 3 above)
Comment (NicolaeGrigoriu) [inline modules/ding_item_viewer/ding_item_viewer.install]: The identifier is build as group name, entity type, entity bundle, view mode. So, the first ting_object means entity_type and the second ting_object means entity_bundle.

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: fcc1a6d9cd053e501e2621cb7d3e37d377356642
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: fcc1a6d9cd053e501e2621cb7d3e37d377356642
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by DDBdrift** [comment]
https://platform.dandigbib.org/issues/1602 | Closed
